### PR TITLE
Allow user to see request list on Account page (Part 1)

### PR DESCRIPTION
**Pull request by yehee**
#### [Allow user to see request list on Account page](#67)

**Checklist**
- [x] Run `npm run lint:fix` or `yarn lint:fix`

**What and how**
- Created/updated API documentation according to [jamescooke/restapidocs](https://github.com/jamescooke/restapidocs/tree/master/examples)
- Updated `README.md` according to [othneildrew/Best-README-Template](https://github.com/othneildrew/Best-README-Template)
- Implemented **CRUD** for `User`, `Business`, `Request`, and `Tag` schema type

**Additional notes**
- ~Contains changes from `tags` and `misc` -- should be reviewed after #63, #69~
- Implementation of `actions`/`reducers` and work of connecting frontend to backend are **still required** to complete #67
- Be aware: **routes are changed from `/api/users` to `/users`!**
- Testing on Postman:
```json
{
	"info": {
		"_postman_id": "b082c4d9-1a42-4222-8381-cbde85418352",
		"name": "ThinkSmall",
		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json"
	},
	"item": [
		{
			"name": "user",
			"item": [
				{
					"name": "localhost:8080/users",
					"request": {
						"method": "GET",
						"header": [],
						"url": {
							"raw": "localhost:8080/users",
							"host": [
								"localhost"
							],
							"port": "8080",
							"path": [
								"users"
							]
						}
					},
					"response": []
				},
				{
					"name": "localhost:8080/user/:id",
					"request": {
						"method": "GET",
						"header": [],
						"url": {
							"raw": "localhost:8080/owner/5ef3e0feb37638721e4dbfcd",
							"host": [
								"localhost"
							],
							"port": "8080",
							"path": [
								"owner",
								"5ef3e0feb37638721e4dbfcd"
							]
						}
					},
					"response": []
				},
				{
					"name": "localhost:8080/user",
					"request": {
						"method": "POST",
						"header": [],
						"body": {
							"mode": "raw",
							"raw": "{\n    \"firstName\": \"John\",\n    \"lastName\": \"Doe\",\n    \"imageURL\": \"https://i.pinimg.com/736x/18/45/d6/1845d65689ecc7cecfaf8f4d958a7f67.jpg\",\n    \"description\": \"Hi! I run a small business in Vancouver. Looking forward to meeting you all!\"\n}",
							"options": {
								"raw": {
									"language": "json"
								}
							}
						},
						"url": {
							"raw": "localhost:8080/user",
							"host": [
								"localhost"
							],
							"port": "8080",
							"path": [
								"user"
							]
						}
					},
					"response": []
				},
				{
					"name": "localhost:8080/user/:id",
					"request": {
						"method": "PUT",
						"header": [],
						"body": {
							"mode": "raw",
							"raw": "{\n    \"businesses\": [0],\n    \"firstName\":\"John\",\n    \"lastName\":\"Doe\",\n    \"imageURL\":\"https://images.unsplash.com/photo-1585110396000-c9ffd4e4b308?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=934&q=80\",\n    \"description\":\"Hi! I run a small business in Vancouver. Looking forward to meeting you all!\"\n}",
							"options": {
								"raw": {
									"language": "json"
								}
							}
						},
						"url": {
							"raw": "localhost:8080/user/5ef3e8b749db9440f8251112",
							"host": [
								"localhost"
							],
							"port": "8080",
							"path": [
								"user",
								"5ef3e8b749db9440f8251112"
							]
						}
					},
					"response": []
				},
				{
					"name": "localhost:8080/user/:id",
					"request": {
						"method": "DELETE",
						"header": [],
						"url": {
							"raw": "localhost:8080/owner/5ef3e8b749db9440f8251185",
							"host": [
								"localhost"
							],
							"port": "8080",
							"path": [
								"owner",
								"5ef3e8b749db9440f8251185"
							]
						}
					},
					"response": []
				}
			],
			"protocolProfileBehavior": {}
		}
	],
	"protocolProfileBehavior": {}
}
```

**Screenshots**
Screen Shot 2020-07-04 at 8 29 02 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,64 @@
...
 
 As a business owner user, the app will need to store the following data: name of the business, business ID (assigned by the application), location, required services, business description, and status of the store (if they require a service or not). As a community volunteer, the following data are necessary: volunteer name, volunteer ID, general location (city), skills that they can provide, and volunteer description (portfolio or link). Additional functionality that we would like to have given that there is time, is the implementation of a point system so that volunteers can collect non-monetary reimbursement in return for work that is completed which could then be used as a store credit at these businesses.
 
-## Project Task Requirements
+### Built with
+* [React](https://reactjs.org/)
+* [Redux](https://redux.js.org/)
+* [Node.js](https://nodejs.org/)
+* [Express](https://expressjs.com/)
+* [AWS](https://aws.amazon.com/)
```
Comment (anhyesun) [inline README.md]: We can specify AWS Amazon S3, and could we add google (maps, OAuth 2.0) here?

Comment (yehee) [reply README.md]: Sure!

File: api/docs/USER.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+Update a single user.
+
+**URL**: `/user/:id`
+
+**URL Parameters**: `id=[int]` where `id` is the ID of the user
+
+**Request body example**
+```json
+{
+    "email": "alice@nwplus.io"
```
Comment (anhyesun) [inline api/docs/USER.md]: could you change the example email with something general like @fake.com?

File: api/docs/USER.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    "data": {
+        "owns": [],
+        "tags": [
+            "Developer"
+        ],
+        "available": true,
+        "_id": "5f04ba889c40d6cbcb63b9c7",
+        "givenName": "Alice",
+        "familyName": "Kim",
+        "email": "alice@nwplus.io",
```
Comment (anhyesun) [inline api/docs/USER.md]: same as above

File: api/models/user.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const { Schema } = mongoose
+
+const User = new Schema(
+    {
+        givenName: { type: String, required: true },
+        familyName: { type: String, required: true },
+        email: { type: String, required: true },
+        imageUrl: String,
+        description: String,
+        supplementaryUrl: String,
```
Comment (anhyesun) [inline api/models/user.js:11]: supplementaryUrl can't found on documentation(api/docs/USER.md) example. What's that?

File: db/user.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        familyName: 'Doe',
+        email: 'jdoe@mail.io',
+        imageUrl: 'https://images.unsplash.com/photo-1585110396000-c9ffd4e4b308?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=934&q=80',
+        description: 'Hi! I run a small business in Vancouver. Looking forward to meeting you all!',
+    }, {
+        givenName: 'Jane',
+        familyName: 'Doe',
+        email: 'jdoe@mail.io',
+        imageUrl: 'https://images.unsplash.com/photo-1529626455594-4ff0802cfb7e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80',
+        supplementaryUrl: 'https://unsplash.com/photos/6W4F62sN_yI',
```
Comment (anhyesun) [inline db/user.js:13]: related with the above comment, I found that example on content example. I just wanna know what it is and how different imageUrl and supplementaryUrl?

**Review by anhyesun — APPROVED** [review]
Looks good to me, and thanks for your work :) I'm just curious,  about folder name `api/routes` is a typo of `routers`? or it has a different meaning?

File: api/models/user.js (hunk 4 above)
Comment (yehee) [reply api/models/user.js:11]: This is meant to be the resume or portfolio link for volunteers. Let me know if you have a better suggestion!

**Comment by yehee** [comment]
> Looks good to me, and thanks for your work :) I'm just curious, about folder name `api/routes` is a typo of `routers`? or it has a different meaning?

`routes` folder was already there when we create `api` folder with `express-generator`. Are you referring to something else?

**Comment by anhyesun** [comment]
You are right, I thought that was a typo. lol sorry for confusing!

**Pull request merged**